models/pancakeswap/ethereum/pancakeswap_v2_ethereum_mmpool_trades.sql

```diff
@@ -0,0 +1,108 @@
+             THEN '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' ELSE basetoken END       AS token_sold_address,     
+        t.contract_address                                                              AS project_contract_address,
+        t.evt_tx_hash                                                                   AS tx_hash,
+        ''                                                                              AS trace_address,
+        t.evt_index
+    FROM {{ source('pancakeswap_v2_ethereum', 'PancakeSwapMMPool_evt_Swap') }} t
+    {% if is_incremental() %}
+    WHERE t.evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+)
+
+SELECT
+    'ethereum'                                                        AS blockchain
+     , 'pancakeswap'                                             AS project
+     , 'mmpool'                                                  AS version
+     , TRY_CAST(date_trunc('DAY', dexs.block_time) AS date)      AS block_date
+     , dexs.block_time
+     , bep20a.symbol                                             AS token_bought_symbol
+     , bep20b.symbol                                             AS token_sold_symbol
+     , case
+           when lower(bep20a.symbol) > lower(bep20b.symbol) then concat(bep20b.symbol, '-', bep20a.symbol)
+           else concat(bep20a.symbol, '-', bep20b.symbol)
+       end                                                       AS token_pair
+     , dexs.token_bought_amount_raw / power(10, bep20a.decimals) AS token_bought_amount
+     , dexs.token_sold_amount_raw / power(10, bep20b.decimals)   AS token_sold_amount
+     , CAST(dexs.token_bought_amount_raw AS DECIMAL(38,0))       AS token_bought_amount_raw
+     , CAST(dexs.token_sold_amount_raw AS DECIMAL(38,0))         AS token_sold_amount_raw
+     , coalesce(
+        dexs.amount_usd
+        , (dexs.token_bought_amount_raw / power(10, p_bought.decimals)) * p_bought.price
+        , (dexs.token_sold_amount_raw / power(10, p_sold.decimals)) * p_sold.price
+        )                                                        AS amount_usd
+     , dexs.token_bought_address
+     , dexs.token_sold_address
+     , coalesce(dexs.taker, tx.from)                             AS taker -- subqueries rely on this COALESCE to avoid redundant joins with the transactions table
+     , dexs.maker
+     , dexs.project_contract_address
+     , dexs.tx_hash
+     , tx.from                                                   AS tx_from
+     , tx.to                                                     AS tx_to
+     , dexs.trace_address
+     , dexs.evt_index
+FROM dexs
+INNER JOIN {{ source('ethereum', 'transactions') }} tx
+    ON tx.hash = dexs.tx_hash
+    {% if not is_incremental() %}
+    AND tx.block_time >= '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+    AND tx.block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+LEFT JOIN {{ ref('tokens_erc20') }} bep20a
```

Comment:
extremely minor, but can we change the alias of tables in the ethereum model(s) to be `erc20` to avoid confusion in future readability?

i figured i'd request this minor change while i wait to merge this, as i'm curious to see what happens when rebuilding these objects and flipping alias names around. since the new rollup view uses the same alias as old spell, which was a table vs. view, i wonder if automated drop/create through dbt works fine.

i will merge this one separate from a larger batch of PRs, to ensure we can monitor closely. otherwise, all looks good!

Reply:
hey @jeff-dude  have committed the change, thanks!